src/path.lisp

```diff
@@ -59,6 +59,18 @@
                 (f w) (f h)))
 
 
+(defun reset-scissors ()
+  (%nanovg:reset-scissor *canvas-handle*))
+
+
+(defmacro with-scissors (origin w h &body body)
+  `(unwind-protect
+        (progn
+          (scissors ,origin ,w ,h)
+          ,@body)
+     (reset-scissors)))
```

Comment:
This is a bit tricky wrapper. If there will be inner WITH-SCISSORS `(with-scissors (...) (with-scissors (...)))`, it will reset the outer one completely. Gonna be quite confusing, especially if inner WITH-SCISSORS used in some sort of function that doesn't expected to be wrapped with outer WITH-SCISSORS. Maybe WITH-SCISSORS should restore previous scissors if that is at all possible.